elements/includes/creative_commons.inc

```diff
@@ -2,29 +2,79 @@
 
 /**
  * @file
- * Code for creative commons form elements.
+ * Code for `creative_commons` form elements.
  */
 
 /**
- * Creates a creative_commons form element.
- *
- * @param array $form_state
- *   The form state.
+ * Constants
+ */
+const XML_FORM_ELEMENTS_CREATIVE_COMMONS_BASE_URL = 'http://creativecommons.org/';
```

Comment:
Last one here. Any reason to stay with the old HTTP instead of HTTPS?

Reply:
@DiegoPino If you remember [comment 385427248](https://github.com/Islandora/islandora_xml_forms/pull/253#issuecomment-385427248), the API considers `https` as invalid (and returns `http` URLs itself). This is unfortunate, as all the links can be reached via SSL, but we have to live with it.

Reply:
I know but it scares me that in 2 years that will be not the case anymore... =(

Reply:
@DiegoPino true, which means we will have to modify then (and mend existing objects - which might actually be the reason for CC to vote against updating the URL thus far).

Perhaps the proper way to deal with this would be to _store_ the non-SSL URL, but to _render_ it as a `https` link.

Reply:
@bondjimbond @rosiel what do you think? I kinda prefer storing good data long-term valid stuff (`https`) and validating against the `HTTP' one for now while doing a hot swap of protocols using `parse_url`or similar?. But can live with any style really.  Is there a proper style? Who said URIs where stable!!!!!! Damn WEB

Reply:
My gut instinct is to stick with HTTP, since that is what the data currently requires, and we can't make assumptions about what CreativeCommons will and won't do. Is there any reason we would expect that HTTP URIs would no longer be valid at some point in the future, while still keeping the equivalent HTTPS URI valid?

If anything, wouldn't they just force redirect you to HTTPS when you visit the site? HTTP links should still be valid long-term as far as I understand.

Reply:
Ok, lets do that then, lets saty on http. Is the CC API versioned? Can we say "this code works for version Blabla .0?"

Reply:
@DiegoPino It is! We are using here the (currently stable) 1.5 version (see https://api.creativecommons.org/docs/). Where would you like that info to be shown?

Reply:
Yep, Ok, will just put that in the release notes. Should be enough right? README could be also a good option. Thanks!

Reply:
@DiegoPino Except for the newly introduced "Known Issues" subsubsection, there is no particular mention of the CC element in the README, right? So, perhaps, putting this info there would be a bit too specific (after all, it is the README of the _huge_ `islandora_xml_forms` module)? But I will not object if you or others insist in putting a note there (but where?)...